src/domain/auth-rebac/domain/src/repos/rebac_repository.rs

```diff
@@ -14,59 +14,60 @@ use crate::{Entity, EntityType, EntityWithRelation, PropertyName, PropertyValue,
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+#[cfg_attr(feature = "testing", mockall::automock)]
 #[async_trait::async_trait]
 pub trait RebacRepository: Send + Sync {
     // Properties
 
-    async fn set_entity_property(
+    async fn set_entity_property<'a>(
         &self,
-        entity: &Entity,
+        entity: &Entity<'a>,
         property_name: PropertyName,
-        property_value: &PropertyValue,
+        property_value: &PropertyValue<'a>,
     ) -> Result<(), SetEntityPropertyError>;
 
-    async fn delete_entity_property(
+    async fn delete_entity_property<'a>(
         &self,
-        entity: &Entity,
+        entity: &Entity<'a>,
         property_name: PropertyName,
     ) -> Result<(), DeleteEntityPropertyError>;
 
-    async fn get_entity_properties(
+    async fn get_entity_properties<'a, 'b>(
         &self,
-        entity: &Entity,
-    ) -> Result<Vec<(PropertyName, PropertyValue)>, GetEntityPropertiesError>;
+        entity: &Entity<'a>,
+    ) -> Result<Vec<(PropertyName, PropertyValue<'b>)>, GetEntityPropertiesError>;
```

Comment:
Try getting rid of these lifetimes, it looks like garbage

Reply:
> Due to the fact that mockall macros play with lifetimes, leading to impenetrable compilation errors, I had to raise the derive here so that the lifetimes are exactly matched up

(c) https://github.com/kamu-data/kamu-cli/pull/761#discussion_r1724684785

--

Alternatively, I can replace the mock-impl with a real one specifically for my test case -- I'll think about what's best to do, because I'm not sure

Reply:
Since the processing of the added dataset has been moved outside of `CreateDatasetFromSnapshotUseCaseImpl`, there is no need for `MockRebacRepository` anymore

Fixed